database/pebble/db.go

```diff
@@ -0,0 +1,326 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package pebble
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/cockroachdb/pebble"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"golang.org/x/exp/slices"
+
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/utils/units"
+)
+
+// pebbleByteOverHead is the number of bytes of constant overhead that
+// should be added to a batch size per operation.
+const pebbleByteOverHead = 8
+
+var (
+	_ database.Database = (*Database)(nil)
+
+	ErrInvalidOperation = errors.New("invalid operation")
+
+	defaultCacheSize = 512 * units.MiB
+	DefaultConfig    = Config{
+		CacheSize:                   defaultCacheSize,
+		BytesPerSync:                512 * units.KiB,
+		WALBytesPerSync:             0, // Default to no background syncing.
+		MemTableStopWritesThreshold: 8,
+		MemTableSize:                defaultCacheSize / 4,
+		MaxOpenFiles:                4096,
+	}
```

Comment:
What is the relation between `CacheSize` and `MemTableSize`? Also should `MemTableSize` be `/ 2` rather than `/ 4` because it seems like there can only be 2 of them actively